diff --git Makefile Makefile
index 3e7161e..4ac6cde 100644
--- Makefile
+++ Makefile
@@ -1,19 +1,7 @@
 # Jivelite makefile
 
 # set PREFIX for location of luajit include and libs
-PREFIX ?= /usr/local
-
-ifeq ($(shell uname -m), aarch64)
-        LUAJIT_VERSION = 2.1
-        export LUAJIT_VERSION
-        CFLAGS += -DLUAJIT_VERSION=21
-	export CFLAGS
-else
-        LUAJIT_VERSION = 2.0
-        export LUAJIT_VERSION
-        CFLAGS += -DLUAJIT_VERSION=20
-	export CFLAGS
-endif
+PREFIX ?= /opt/jivelite
 
 all: srcs libs
 
diff --git lib-src/Makefile lib-src/Makefile
index fc643e4..858dd61 100644
--- lib-src/Makefile
+++ lib-src/Makefile
@@ -2,8 +2,8 @@
 # NB - repo has modified paths which assume use of luajit2.0 from source
 
 TARGET   ?= linux
-
-LD_FLAGS ?= "-shared"
+LD_FLAGS ?= "-shared -Wl,-rpath,$(PREFIX)/lib"
+CFLAGS ?= "-s"
 
 LIBS = cjson luaexpat luafilesystem luamd5 luasocket luazipfilter
 
diff --git lib-src/lua-cjson-2.1.0/Makefile lib-src/lua-cjson-2.1.0/Makefile
index 2b14746..cc432ba 100644
--- lib-src/lua-cjson-2.1.0/Makefile
+++ lib-src/lua-cjson-2.1.0/Makefile
@@ -13,12 +13,12 @@
 ##### Build defaults #####
 LUA_VERSION =       5.1
 TARGET =            cjson.so
-PREFIX ?=           /usr/local
+PREFIX =            /opt/jivelite
 #CFLAGS =            -g -Wall -pedantic -fno-inline
-CFLAGS +=            -O3 -Wall -pedantic -DNDEBUG
+CFLAGS =            -s -O3 -Wall -pedantic -DNDEBUG
 CJSON_CFLAGS =      -fpic
 #CJSON_LDFLAGS =     -shared
-LUA_INCLUDE_DIR =   $(PREFIX)/include/luajit-$(LUAJIT_VERSION)
+LUA_INCLUDE_DIR =   $(PREFIX)/include
 LUA_CMODULE_DIR =   $(PREFIX)/lilu$(LUA_VERSION)
 LUA_MODULE_DIR =    $(PREFIX)/share/lu$(LUA_VERSION)
 LUA_BIN_DIR =       $(PREFIX)/bin
diff --git lib-src/luaexpat-1.2.0/config lib-src/luaexpat-1.2.0/config
index 4f94f31..5474c8b 100644
--- lib-src/luaexpat-1.2.0/config
+++ lib-src/luaexpat-1.2.0/config
@@ -4,7 +4,7 @@ LUA_LIBDIR= $(PREFIX)/lilu5.1
 # System's lua directory (where Lua libraries are installed)
 LUA_DIR= $(PREFIX)/share/lu5.1
 # Lua includes directory
-LUA_INC= $(PREFIX)/include/luajit-$(LUAJIT_VERSION)
+LUA_INC= $(PREFIX)/include
 
 # OS dependent
 #LIB_OPTION= -shared #for Linux
@@ -29,5 +29,5 @@ CWARNS = -Wall -pedantic \
         -Wshadow \
         -Wwrite-strings
 
-CFLAGS += $(CWARNS) -ansi -O2 -I$(LUA_INC) -fPIC
+CFLAGS = $(CWARNS) -ansi -O2 -I$(LUA_INC) -fPIC
 CC = gcc
diff --git lib-src/luafilesystem-1.6.2/config lib-src/luafilesystem-1.6.2/config
index 06c8610..69a9959 100644
--- lib-src/luafilesystem-1.6.2/config
+++ lib-src/luafilesystem-1.6.2/config
@@ -1,13 +1,13 @@
 # Installation directories
 
 # Default installation prefix
-PREFIX ?= /usr/local
+PREFIX=/opt/jivelite
 
 # System's libraries directory (where binary libraries are installed)
 LUA_LIBDIR= $(PREFIX)/lilu5.1
 
 # Lua includes directory
-LUA_INC= $(PREFIX)/include/luajit-$(LUAJIT_VERSION)
+LUA_INC= $(PREFIX)/include
 
 # OS dependent
 #LIB_OPTION= -shared #for Linux
@@ -16,7 +16,7 @@ LUA_INC= $(PREFIX)/include/luajit-$(LUAJIT_VERSION)
 LIBNAME= $T.so.$V
 
 # Compilation directives
-WARN= -O2 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -pedantic
+WARN= -s -O3 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -pedantic
 INCS= -I$(LUA_INC)
 CFLAGS= $(WARN) $(INCS)
 CC= gcc
diff --git lib-src/luamd5/Makefile lib-src/luamd5/Makefile
index ff2eecb..43476d0 100644
--- lib-src/luamd5/Makefile
+++ lib-src/luamd5/Makefile
@@ -2,7 +2,7 @@
 
 # change these to reflect your Lua installation
 #LUA= /tmp/lhf/lua-5.1.4
-LUAINC= $(PREFIX)/include/luajit-$(LUAJIT_VERSION)
+LUAINC= $(PREFIX)/include
 LUALIB= $(LUA)/lib
 LUABIN= $(LUA)/bin
 
@@ -74,7 +74,7 @@ MD5OBJ= $(MD5LIB)
 
 # probably no need to change anything below here
 CC= gcc
-CFLAGS += $(INCS) $(DEFS) $(WARN) -O2 $G -fPIC
+CFLAGS= $(INCS) $(DEFS) $(WARN) -O2 $G -fPIC
 WARN= #-ansi -pedantic -Wall
 INCS= -I$(LUAINC) $(MD5INC)
 
diff --git lib-src/luasocket-2.0.2/config lib-src/luasocket-2.0.2/config
index 1952a57..b5788c1 100644
--- lib-src/luasocket-2.0.2/config
+++ lib-src/luasocket-2.0.2/config
@@ -18,7 +18,7 @@ UNIX_SO=unix.$(EXT)
 #LUAINC=-I/usr/local/include/lua50
 #LUAINC=-I/usr/local/include/lua5.1
 #LUAINC=-Ilua-5.1.1/src
-LUAINC=-I$(PREFIX)/include/luajit-$(LUAJIT_VERSION)
+LUAINC=-I$(PREFIX)/include
 
 #------
 # Compat-5.1 directory
diff --git lib-src/luazipfilter/Makefile lib-src/luazipfilter/Makefile
index 70dcfd8..ebcc485 100644
--- lib-src/luazipfilter/Makefile
+++ lib-src/luazipfilter/Makefile
@@ -1,5 +1,5 @@
-CFLAGS  += -O2 -I$(PREFIX)/include/luajit-$(LUAJIT_VERSION) -fPIC
-LDFLAGS = $(MYLDFLAGS) -lz
+CFLAGS  += -I$(PREFIX)/include -fPIC -s -O3
+LDFLAGS = $(MYLDFLAGS) -lz -L$(PREFIX)/lib
 
 SOURCES = luazipfilter.c
 LIB = zipfilter.so
diff --git patches/jivelite-lua-5.1.patch patches/jivelite-lua-5.1.patch
index f3fdcf1..0ccc67a 100644
--- patches/jivelite-lua-5.1.patch
+++ patches/jivelite-lua-5.1.patch
@@ -1,152 +0,0 @@
-Index: Makefile
-===================================================================
---- Makefile	(revision 784)
-+++ Makefile	(working copy)
-@@ -1,7 +1,7 @@
- # Jivelite makefile
- 
- # set PREFIX for location of luajit include and libs
--PREFIX ?= /usr/local
-+PREFIX ?= /opt/jivelite
- 
- all: srcs libs
- 
-Index: lib-src/Makefile
-===================================================================
---- lib-src/Makefile	(revision 784)
-+++ lib-src/Makefile	(working copy)
-@@ -2,7 +2,8 @@
- # NB - repo has modified paths which assume use of luajit2.0 from source
- 
- TARGET   ?= linux
--LD_FLAGS ?= "-shared"
-+LD_FLAGS ?= "-shared -Wl,-rpath,$(PREFIX)/lib"
-+CFLAGS ?= "-s"
- 
- LIBS = cjson luaexpat luafilesystem luamd5 luasocket luazipfilter
- 
-Index: lib-src/lua-cjson-2.1.0/Makefile
-===================================================================
---- lib-src/lua-cjson-2.1.0/Makefile	(revision 784)
-+++ lib-src/lua-cjson-2.1.0/Makefile	(working copy)
-@@ -13,12 +13,12 @@
- ##### Build defaults #####
- LUA_VERSION =       5.1
- TARGET =            cjson.so
--PREFIX =            /usr/local
-+PREFIX =            /opt/jivelite
- #CFLAGS =            -g -Wall -pedantic -fno-inline
--CFLAGS =            -O3 -Wall -pedantic -DNDEBUG
-+CFLAGS =            -s -O3 -Wall -pedantic -DNDEBUG
- CJSON_CFLAGS =      -fpic
- #CJSON_LDFLAGS =     -shared
--LUA_INCLUDE_DIR =   $(PREFIX)/include/luajit-2.0
-+LUA_INCLUDE_DIR =   $(PREFIX)/include
- LUA_CMODULE_DIR =   $(PREFIX)/lilu$(LUA_VERSION)
- LUA_MODULE_DIR =    $(PREFIX)/share/lu$(LUA_VERSION)
- LUA_BIN_DIR =       $(PREFIX)/bin
-Index: lib-src/luaexpat-1.2.0/config
-===================================================================
---- lib-src/luaexpat-1.2.0/config	(revision 784)
-+++ lib-src/luaexpat-1.2.0/config	(working copy)
-@@ -4,7 +4,7 @@
- # System's lua directory (where Lua libraries are installed)
- LUA_DIR= $(PREFIX)/share/lu5.1
- # Lua includes directory
--LUA_INC= $(PREFIX)/include/luajit-2.0
-+LUA_INC= $(PREFIX)/include
- 
- # OS dependent
- #LIB_OPTION= -shared #for Linux
-Index: lib-src/luafilesystem-1.6.2/config
-===================================================================
---- lib-src/luafilesystem-1.6.2/config	(revision 784)
-+++ lib-src/luafilesystem-1.6.2/config	(working copy)
-@@ -1,13 +1,13 @@
- # Installation directories
- 
- # Default installation prefix
--PREFIX=/usr/local
-+PREFIX=/opt/jivelite
- 
- # System's libraries directory (where binary libraries are installed)
--LUA_LIBDIR= $(PREFIX)/lilu5.1
-+LUA_LIBDIR= $(PREFIX)/lilu5.1
- 
- # Lua includes directory
--LUA_INC= $(PREFIX)/include/luajit-2.0
-+LUA_INC= $(PREFIX)/include
- 
- # OS dependent
- #LIB_OPTION= -shared #for Linux
-@@ -16,7 +16,7 @@
- LIBNAME= $T.so.$V
- 
- # Compilation directives
--WARN= -O2 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -pedantic
-+WARN= -s -O3 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -pedantic
- INCS= -I$(LUA_INC)
- CFLAGS= $(WARN) $(INCS)
- CC= gcc
-Index: lib-src/luamd5/Makefile
-===================================================================
---- lib-src/luamd5/Makefile	(revision 784)
-+++ lib-src/luamd5/Makefile	(working copy)
-@@ -2,7 +2,7 @@
- 
- # change these to reflect your Lua installation
- #LUA= /tmp/lhf/lua-5.1.4
--LUAINC= $(PREFIX)/include/luajit-2.0
-+LUAINC= $(PREFIX)/include
- LUALIB= $(LUA)/lib
- LUABIN= $(LUA)/bin
- 
-Index: lib-src/luasocket-2.0.2/config
-===================================================================
---- lib-src/luasocket-2.0.2/config	(revision 784)
-+++ lib-src/luasocket-2.0.2/config	(working copy)
-@@ -18,7 +18,7 @@
- #LUAINC=-I/usr/local/include/lua50
- #LUAINC=-I/usr/local/include/lua5.1
- #LUAINC=-Ilua-5.1.1/src
--LUAINC=-I$(PREFIX)/include/luajit-2.0
-+LUAINC=-I$(PREFIX)/include
- 
- #------
- # Compat-5.1 directory
-Index: lib-src/luazipfilter/Makefile
-===================================================================
---- lib-src/luazipfilter/Makefile	(revision 784)
-+++ lib-src/luazipfilter/Makefile	(working copy)
-@@ -1,5 +1,5 @@
--CFLAGS  = -O2 -I$(PREFIX)/include/luajit-2.0 -fPIC
--LDFLAGS = $(MYLDFLAGS) -lz
-+CFLAGS  = -I$(PREFIX)/include -fPIC -s -O3
-+LDFLAGS = $(MYLDFLAGS) -lz -L$(PREFIX)/lib
- 
- SOURCES = luazipfilter.c
- LIB = zipfilter.so
-Index: src/Makefile
-===================================================================
---- src/Makefile	(revision 784)
-+++ src/Makefile	(working copy)
-@@ -1,8 +1,8 @@
- LDFLAGS ?= -lrt
- SOURCES ?= platform_linux.c
- 
--CFLAGS  += -I. -I$(PREFIX)/include/luajit-2.0 -I/usr/include/SDL -Wall -fPIC
--LDFLAGS += -lSDL -lSDL_ttf -lSDL_image -lSDL_gfx -lluajit-5.1 -lm -lpthread
-+CFLAGS  += -I. -I$(PREFIX)/include -I/usr/include/SDL -Wall -fPIC -DOPTJIVELITE -s -O3
-+LDFLAGS +=-lSDL -lSDL_ttf -lSDL_image -lSDL_gfx -L$(PREFIX)/lib -llua -lm -lpthread
- EXE = ../bin/jivelite
- 
- DEPS    = jive.h common.h log.h version.h
-@@ -17,7 +17,7 @@
- 	cd visualizer; make
- 
- $(EXE): $(OBJECTS)
--	$(CC) $(OBJECTS) $(LDFLAGS) -o $@
-+	$(CC) $(OBJECTS) -Wl,-rpath,$(PREFIX)/lib $(LDFLAGS) -o $@
- 
- $(OBJECTS): $(DEPS)
- 
diff --git src/Makefile src/Makefile
index b0cbc50..445ba03 100644
--- src/Makefile
+++ src/Makefile
@@ -1,8 +1,8 @@
 LDFLAGS ?= -lrt
 SOURCES ?= platform_linux.c
 
-CFLAGS  += -I. -I$(PREFIX)/include/luajit-$(LUAJIT_VERSION) -I/usr/include/SDL -Wall -fPIC
-LDFLAGS += -lSDL -lSDL_ttf -lSDL_image -lSDL_gfx -lluajit-5.1 -lm -lpthread
+CFLAGS  += -I. -I$(PREFIX)/include -I$(PREFIX)/include/SDL -Wall -fPIC -DOPTJIVELITE -s -O3
+LDFLAGS +=-lSDL -lSDL_ttf -lSDL_image -lSDL_gfx -L$(PREFIX)/lib -llua -lm -lpthread
 EXE = ../bin/jivelite
 
 DEPS    = jive.h common.h log.h version.h
@@ -17,12 +17,12 @@ visualizer:
 	cd visualizer; make
 
 $(EXE): $(OBJECTS)
-	$(CC) $(OBJECTS) $(LDFLAGS) -o $@
+	$(CC) $(OBJECTS) -o $@ -Wl,-rpath,$(PREFIX)/lib,-E $(LDFLAGS)
 
 $(OBJECTS): $(DEPS)
 
 .c.o:
-	$(CC) $(CFLAGS) $< -c -o $@
+	$(CC) -c -o $@ $(CFLAGS) $<
 
 clean:
 	rm -f $(OBJECTS) $(EXE)
